src/Aspect/ContractCheckerAspect.php

```diff
@@ -32,6 +32,11 @@ class ContractCheckerAspect implements Aspect
     private $reader = null;
 
     /**
+     * @var MethodInvocation
+     */
+    private $invocation;
```

Comment:
I created private property to have access to MethodInvocation instance in ContractCheckerAspect::170 line. I don't want to create new parameter in isContractSatisfied method for this (it would be 5th parameter). 

Why do we throw exception here? If we return false instead of raising exception, then we lose previous exception (for example - assertion exeption) details.